hastexo/hastexo.py

```diff
@@ -658,6 +658,10 @@ def launch_new_window(self, request, suffix=''):
         """
         The fullscreen lab view, opened in a new browser window.
         """
+
+        if not request.cookies.get('edxloggedin', False):
```

Comment:
Just checking: are you sure you want to test for the existence of one specific cookie here, rather than generically checking whether the request is associated with a valid session? Does `request.session` give you a generic way to do that? (I'm not sure if it does.)

Reply:
No, there is no session option, it's a [webob.Request](https://docs.pylonsproject.org/projects/webob/en/stable/api/request.html#module-webob.request). We can check for `edx-user-info`, `edxloggedin` or `sessionid` cookies. I can switch to checking if `sessionid` is present.

Reply:
Maybe go for `sessionid`, then. Who knows if `edxloggedin` doesn't become `openedxloggedin` at some point. :)

Reply:
Ack, done.